package.json

```diff
@@ -18,19 +18,20 @@
   "types": "lib/peg.d.ts",
   "bin": "bin/peggy",
   "scripts": {
-    "pretest": "npm run parser",
-    "test": "npm run test:bare",
-    "test:bare": "mocha --recursive test --exclude test/browser.stub.js --reporter min",
+    "clean": "rimraf -f build && rimraf -f browser && mkdir build && cd build && mkdir ts && mkdir rollup && mkdir terser && cd .. && mkdir browser",
+    "terser": "terser -c passes=2 -m --sequences 40 build/rollup/peg.umd.js > build/terser/peggy.umd.min.js && terser -c passes=2 -m --sequences 40 --module build/rollup/peg.esmodule.js > build/terser/peggy.esmodule.min.js",
+    "rollup": "rollup -c",
+    "ts": "tsc --build tsconfig.json",
+    "set_version": "node -e \"const fs = require('fs'), pkg = JSON.parse(fs.readFileSync('./package.json')); fs.writeFileSync('./lib/version.js', '\\n\\'use strict\\';\\n\\nmodule.exports = \\'' + pkg.version + '\\';\\n');\"",
```

Comment:
`tools/`, but yes.

Reply:
> That is much more maintainable.

These things have been studied, and it actually isn't.

I'm willing to do what you guys want, but please don't make infinitive claims like this without evidence.  No, spreading code across multiple files in multiple directories, to be invoked by a runner somewhere else, which requires you to know one project's specific and non-idiomatic directory structure, instead of putting it in the standard place where everything goes, is not "more maintainable."

You yourselves put much more complex things in here, when it's babel configurations.

Reply:
The babel config was my fault, not Mingun's.  And it was a stopgap to get us to this point, hated it. :)

Reply:
> It is better to explicitly move that script to the bin/update-version.js file. That is much more maintainable.

Done

Reply:
> instead of putting it in the standard place where everything goes

Still, I can not fail to notice that the presence of a code in the form of a line in a place that is not checked by any tools cannot be called idiomatic even with a tension... Well, never-mind

Reply:
> Still, I can not fail to notice that the presence of a code in the form of a line in a place that is not checked by any tools cannot be called idiomatic even with a tension... Well, never-mind

It remains the community standard, and there's good reason for it.

Idiom means "what everyone else does," not "what we prefer."